src/masternodes/mn_rpc.cpp

```diff
@@ -128,35 +199,128 @@ std::vector<CTxIn> GetInputs(UniValue const& inputs) {
-        return GetInputs(explicitInputs.get_array());
-    } else {
-        std::vector<COutput> vecOutputs;
-        CCoinControl cctl;
-        cctl.m_avoid_address_reuse = false;
-        cctl.m_min_depth = 1;
-        cctl.m_max_depth = 999999999;
-        cctl.matchDestination = auth;
-        cctl.m_tokenFilter = {DCT_ID{0}};
-
-        pwallet->BlockUntilSyncedToCurrentChain();
-        auto locked_chain = pwallet->chain().lock();
-        LOCK(pwallet->cs_wallet);
+boost::optional<CScript> AmIFounder(CWallet* const pwallet) {
+    for(auto const & script : Params().GetConsensus().foundationMembers) {
+        if(IsMine(*pwallet, script) == ISMINE_SPENDABLE)
+            return { script };
+    }
+    return {};
+}
+
+static boost::optional<CTxIn> GetAuthInputOnly(CWallet* const pwallet, CTxDestination const& auth) {
+
+    std::vector<COutput> vecOutputs;
+    CCoinControl cctl;
+    cctl.m_avoid_address_reuse = false;
+    cctl.m_min_depth = 1;
+    cctl.m_max_depth = 999999999;
+    cctl.matchDestination = auth;
+    cctl.m_tokenFilter = {DCT_ID{0}};
 
-        pwallet->AvailableCoins(*locked_chain, vecOutputs, true, &cctl, 1, MAX_MONEY, MAX_MONEY, 1);
+    auto locked_chain = pwallet->chain().lock();
+    LOCK(pwallet->cs_wallet);
 
-        if (vecOutputs.empty()) {
-            throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, strprintf(
-                    "Can't find any UTXO's for owner. Are you an owner? If so, send some coins to address %s and try again!",
-                    EncodeDestination(auth)));
+    pwallet->AvailableCoins(*locked_chain, vecOutputs, true, &cctl, 1, MAX_MONEY, MAX_MONEY, 1);
+
+    if (vecOutputs.empty()) {
+        return {};
+    }
+    return { CTxIn(vecOutputs[0].tx->GetHash(), vecOutputs[0].i) };
+}
+
+CTransactionRef CreateAuthTx(CWallet* const pwallet, std::set<CScript> const & auths, int32_t txVersion) {
+    CMutableTransaction mtx(txVersion);
+
+    // may be use this rates instead of ::dustRelayFee???
+//    FeeCalculation feeCalc;
+//    CFeeRate nFeeRateNeeded = GetMinimumFeeRate(*pwallet, {}, &feeCalc);
+
+    // create tx with one dust output per script
+    for (auto const & auth : auths) {
+        CTxOut authOut(CAmount(1), auth);
+        authOut.nValue = GetDustThreshold(authOut, mtx.nVersion, ::dustRelayFee);
+        mtx.vout.push_back(authOut);
+    }
+
+    return fund(mtx, pwallet, {}, true /*lockUnspents*/), sign(mtx, pwallet, {});
+}
+
+static boost::optional<CTxIn> GetAnyFoundationAuthInput(CWallet* const pwallet) {
+    CScript anyFounder;
+    for (auto const & founderScript : Params().GetConsensus().foundationMembers) {
+        if (IsMine(*pwallet, founderScript) == ISMINE_SPENDABLE) {
+            CTxDestination destination;
+            if (ExtractDestination(founderScript, destination)) {
+                anyFounder = founderScript;
+                auto auth = GetAuthInputOnly(pwallet, destination);
+                if (auth) {
+                    return auth;
```

Comment:
I never use assignment (declaration) inside ifs to avoid possible misreading or any side effects of type conversions. But.. as you wish. Done
... And we can remove extra {} in one-statement blocks, and place them on the same line. Such a great economy of lines! :)

Reply:
@bvbfan not a good idea to declare variables inside condition. No side effects. 

Number of lines should not matter. Most sane linters would have prevented that.